File: models/domain.go
package models

// Domain model
type Domain struct {
	Domain           string `json:"domain"`
	ServersChanged   string `json:"servers_changed"`
	SslGrade         string `json:"ssl_grade"`
	PreviousSslGrade string `json:"previous_ssl_grade"`
	Logo             string `json:"logo"`
	IsDown           string `json:"is_down"`
	Servers          []struct {
		Address  string `json:"address"`
		SslGrade string `json:"ssl_grade"`
		Country  string `json:"country"`
		Owner    string `json:"owner"`
	} `json:"servers"`
}

File: ui/src/interfaces/domain.ts
export interface Domain {
  domain: string;
  serversChanged: boolean;
  sslGrade: string;
  previousSslGrade: string;
  logo: string;
  title: string;
  isDown: boolean;
  servers: Server[];
}

export interface DomainResponse {
  items: Domain[];
}

export interface Server {
  address: string;
  sslGrade: string;
  country: string;
  owner: string;
}
